### ISSUE TITLE
For Turbo compatibility, haml needs a default_format method

### ISSUE THREAD

**Issue by lazylester** [haml/haml#1143]
In a nutshell... a turbo-enabled rails app doesn't respond with the correct content_type header when rendering a haml file.

I finally solved my Rails Turbo problem when I ran across this issue in the turbo-rails repo: https://github.com/hotwired/turbo-rails/issues/287#issue-1085950957

The workaround suggested in this response https://github.com/hotwired/turbo-rails/issues/287#issuecomment-1105729895 works, but only b/c I'm using haml 5.2.2. The Plugin class in which the workaround is monkey-patched doesn't seem to be present in haml 6.0.

Basically within the actionview template_renderer, the handler needs to respond to default_format method. (see the link to the actionview code in the workaround cited above).

**Comment by k0kubun** [comment]
v5.2.2 did not have a `default_format` method either https://github.com/haml/haml/blob/v5.2.2/lib/haml/plugin.rb. According to https://github.com/hotwired/turbo-rails/issues/287#issuecomment-1105729895, it seems like it suffices to define `#default_format` on `Haml::RailsTemplate` because it's essentially `Hamlit::RailsTemplate`.

**Comment by lazylester** [comment]
OK, yes, I'm sure you're correct about v5.2.2. But I think it's still a good idea to include it in v6.0.

**Comment by k0kubun** [comment]
Haml::RailsTemplate is a constant introduced in 6. Have you tried what I said?

**Comment by lazylester** [comment]
Thank you @k0kubun. I understood your reply to be an explanation not a suggestion. Now that I have upgraded to haml v6 I have tried your suggestion and I can confirm that it works. Thank you.

It seems to me that compatibility with Rails Turbo would motivate inclusion of this into haml v6 itself, rather than having to work around with a monkey patch. You see, it was working before (not exactly sure which version of haml, but you have said `default_format` is not supported in 5.2.2, so I must have been using an earlier version). It seems like such a low-risk feature to add to haml v6 that may smooth the migration for other users.

Of course, this only arises b/c I have been naming my files like `filename.haml` iso `filename.html.haml` and perhaps this was not correct, even though it worked fine for many years until Turbo arrived.

Thank you for your help, and thank you for hamlit. Both are truly appreciated.
